elide-async/src/main/java/com/yahoo/elide/async/service/DefaultAsyncQueryDAO.java

```diff
@@ -127,8 +149,38 @@ public AsyncQueryResult setAsyncQueryAndResult(Integer status, String responseBo
         return queryResultObj;
     }
 
-    @Override
-    public Object executeInTransaction(DataStore dataStore, Transactional action) {
+    @SuppressWarnings("unchecked")
+    public Collection<AsyncQuery> loadQueries(String filterExpression) {
+        EntityDictionary dictionary = elide.getElideSettings().getDictionary();
+        RSQLFilterDialect filterParser = new RSQLFilterDialect(dictionary);
+
+        Collection<AsyncQuery> loaded = (Collection<AsyncQuery>) executeInTransaction(dataStore, (tx, scope) -> {
+            try {
+                FilterExpression filter = filterParser.parseFilterExpression(filterExpression, AsyncQuery.class, false);
```

Comment:
This one line can be moved outside the transaction.

Reply:
If we move this line outside the anonymous function - we would have to put it in a try catch block and move the filter variable as a class level variable else we get: Local variable filter defined in an enclosing scope must be final or effectively final. Do you want this variable to be a class level variable?